### PR TITLE
Feature history rebase

### PR DESCRIPTION
**Pull request by greatfighter**
Merge finished history page to develop branch

### CODE REVIEW

File: app/src/main/java/com/hydrofish/app/ui/composables/tabs/HistoryScreen.kt
Comment (marklhj-lee) [inline app/src/main/java/com/hydrofish/app/ui/composables/tabs/HistoryScreen.kt:1]: Move HydroEntry stuff into either utils or some other folder.
Change fetchHydrationData by moving Move Retrofit into ApliClientfile.
After that call can be reduce into val call = ApliClient.apiService.getHydrationData(headers)
You might wanna refactor rest into view model if you get a chance but it's not necessary.

**Review by marklhj-lee — CHANGES_REQUESTED** [review]
Message me if you are confused.

**Review by KevinZxy1273 — APPROVED** [review]
Looks good now

**Review by marklhj-lee — APPROVED** [review]
LGTM

**Review by KevinZxy1273 — APPROVED** [review]
Looks good now

**Review by ArthurQiangLi — APPROVED** [review]
Reviewed

**Pull request merged**